woocommerce/woocommerce-ios
# Orders badge mark IV: UI and integration

Fixes #1089 

In action:
<img src="https://user-images.githubusercontent.com/2722505/62231855-d9ab7380-b392-11e9-9165-3d848120ac36.gif" width="350"/>

And a static screenshot on iPhone X:
<img src="https://user-images.githubusercontent.com/2722505/62231926-ffd11380-b392-11e9-9f98-b071b4e77a25.png" width="350"/>

This PR contains a fix to the Network layer. When I first implemented it, I didn't realise that the network request would be tunnelled, so the payload of the response is different to what I was expecting yesterday. I've updated the Mapper and the mock data to reflect the changes, and that produces a pretty large PR. Sorry... 😊 

## Changes
- Update OrderCountMapper, the mocked json response, and OrdersRemoteTests to fix the payload
- Extracted the code handling the badge in the notifications tab to a separate class. I shouldn't have touched that within the scope of this PR, but it happened.
- Added a view model to encapsulate the logic related to the orders badge. Will move the logic associated to the notifications badge to it when renaming that tab
- Added OrdersBadgeController to manage the actual badge, insert it and remove it from the UI
- Added a similar controller for the notifications badge

## Testing
- Checkout the branch run the tests
- Change the status of orders: try to reach zero orders processing and check that the badge gets removed. Go over 9 and check that the badge reads as "9+"
- Try to touch all the cases that might come to your mind where the badge should bet updated. I am sure I have missed some.


Update release notes:
